Tools/PIDML/kaon_pid.cxx

```diff
@@ -0,0 +1,138 @@
+  Configurable<float> ConfMaxPtCut{"ConfMaxPtCut", 3.0, "Max Pt cut"};
+  Configurable<float> ConfMinPtCut{"ConfMinPtCut", 0.5, "Min Pt cut"};
+  Configurable<float> ConfMinNSigmaTPCCut{"ConfMinNSigmaTPCCut", 3., "N-sigma TPC cut"};
+  Configurable<float> ConfChargeCut{"ConfChargeCut", 0., "N-sigma TPC cut"};
+  Configurable<std::string> cfgPathLocal{"local-path", "/home/shira/kaon-mlpid/", "base path to the local directory with ONNX models"};
+  Configurable<std::string> cfgPathCCDB{"ccdb-path", "Users/m/mkabus/PIDML", "base path to the CCDB directory with ONNX models"};
+  Configurable<std::string> cfgCCDBURL{"ccdb-url", "http://alice-ccdb.cern.ch", "URL of the CCDB repository"};
+  Configurable<int> CfgPid{"pid", 321, "PID to predict"};
+  Configurable<double> CfgCertainty{"certainty", 0.5, "Minimum certainty above which the model accepts a particular type of particle"};
+  Configurable<uint32_t> cfgDetector{"detector", kTPCTOFTRD, "What detectors to use: 0: TPC only, 1: TPC + TOF, 2: TPC + TOF + TRD"};
+  Configurable<uint32_t> cfgTimestamp{"timestamp", -8.0, "Fixed timestamp"};
+  Configurable<bool> cfgUseCCDB{"useCCDB", false, "Whether to autofetch ML model from CCDB. If false, local file will be used."};
+
+  o2::ccdb::CcdbApi ccdbApi;
+
+  Filter collisionFilter = (nabs(aod::collision::posZ) < ConfZvtxCut);
+  Filter trackFilter = (nabs(aod::track::eta) < ConfEtaCut) && (aod::track::pt > ConfMinPtCut) && (aod::track::pt < ConfMaxPtCut);
+
+  // Applying filters
+  using MyFilteredCollisions = soa::Filtered<o2::aod::MyCollisions>;
+  using MyFilteredCollision = MyFilteredCollisions::iterator;
+
+  Partition<o2::aod::MyTracks> positive = (nabs(aod::track::eta) < ConfEtaCut) && (aod::track::pt > ConfMinPtCut) && (aod::track::pt < ConfMaxPtCut) && (aod::track::signed1Pt > ConfChargeCut);
+  Partition<o2::aod::MyTracks> negative = (nabs(aod::track::eta) < ConfEtaCut) && (aod::track::pt > ConfMinPtCut) && (aod::track::pt < ConfMaxPtCut) && (aod::track::signed1Pt < ConfChargeCut);
+
+  void init(o2::framework::InitContext&)
+  {
+    AxisSpec vtxZAxis = {100, -20, 20};
+    std::vector<double> ptBinning = {0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.8, 2.0, 2.2, 2.4, 2.8, 3.2, 3.6, 4.};
+    AxisSpec ptAxis = {ptBinning, "#it{p}_{T} (GeV/#it{c})"};
+
+    if (cfgUseCCDB) {
+      ccdbApi.init(cfgCCDBURL); // Initializes ccdbApi when cfgUseCCDB is set to 'true'
+    }
+    pidModel = new PidONNXModel(cfgPathLocal.value, cfgPathCCDB.value, cfgUseCCDB.value, ccdbApi, cfgTimestamp.value, cfgPid.value, static_cast<PidMLDetector>(cfgDetector.value), cfgCertainty.value);
+
+    histos.add("hChargePos", ";z;", kTH1F, {{3, -1.5, 1.5}});
+    histos.add("hChargeNeg", ";z;", kTH1F, {{3, -1.5, 1.5}});
+    histos.add("hInvariantMass", ";M_{k^{+}k^{-}} (GeV/#it{c}^{2});", kTH1F, {{100, 0., 2.}});
+    histos.add("hPvsdEdxKaon", ";P_{K^{+}K^{-}}; dE/dx in TPC (keV/cm)", kTH2F, {{100, 0., 4.}, {200, 20., 400.}});
```

Comment:
The histogram name is a bit unreadable for me though I cannot find a quick alternative now.